libs/template/experimental/viewport-prio/src/lib/viewport-strategy.directive.ts

```diff
@@ -0,0 +1,155 @@
+import {
+  Directive,
+  ElementRef,
+  Inject,
+  Input,
+  OnDestroy,
+  OnInit,
+  Optional,
+} from '@angular/core';
+import { RxStrategyProvider } from '@rx-angular/cdk/render-strategies';
+import { coerceObservableWith } from '@rx-angular/cdk/coercing';
+import { RxNotification } from '@rx-angular/cdk/notifications';
+import { LetDirective } from '@rx-angular/template/let';
+import {
+  BehaviorSubject,
+  combineLatest,
+  Observable,
+  of,
+  Subject,
+  Subscription,
+} from 'rxjs';
+import { filter, map, mergeAll, withLatestFrom } from 'rxjs/operators';
+import { saveCreateIntersectionObserver } from './intersection-observer';
+
+type IntersectionRatio = Pick<IntersectionObserverEntry, 'intersectionRatio'>;
+
+@Directive({
+  // eslint-disable-next-line @angular-eslint/directive-selector
+  selector: '[viewportStrategy]',
+})
+export class ViewportStrategyDirective implements OnInit, OnDestroy {
+  subscription = new Subscription();
+  /* internal */
+  // Note that we're picking only the `intersectionRatio` property
+  // since this is the only property that we're intersted in.
+  private entriesSubject = new Subject<IntersectionRatio[]>();
+
+  /* internal */
+  private entries$: Observable<IntersectionRatio> = this.entriesSubject.pipe(mergeAll());
+
+  /* internal */
+  private viewportPrioObservables = new BehaviorSubject<
+    Observable<string> | string
+  >(of('noop'));
+
+  /* internal */
+  private internalViewportPrio = this.viewportPrioObservables.pipe(
+    coerceObservableWith(),
+    mergeAll(),
+    // @TODO add comment
+    map((v) => (!v ? 'noop' : v))
+  );
+
+  /* internal */
+  // The directives host intersection observer
+  private observer: IntersectionObserver | null =
+    saveCreateIntersectionObserver((entries) => {
+      this.entriesSubject.next(entries);
+    });
+
+  /* internal */
+  // A stream of visibility events from the host element
+  private visibilityState$ = this.entries$.pipe(
+    map((entry) => {
+      if (entry.intersectionRatio > 0) {
+        return true;
+      } else {
+        return false;
+      }
+    })
+  );
+
+  @Input('viewportPrio')
+  set viewportPrio(prio: string | Observable<string>) {
+    this.viewportPrioObservables.next(prio);
+  }
+
+  constructor(
+    private readonly el: ElementRef<HTMLElement>,
+    private strategyProvider: RxStrategyProvider,
+    @Inject(LetDirective)
+    @Optional()
+    private letDirective: LetDirective<any> | null
+  ) {}
+
+  /* internal */
+  ngOnInit() {
+    const letStrategyName$ = this.strategyProvider.primaryStrategy$.pipe(
+      map(({ name }) => name)
+    );
+
+    let lastNextNotification: RxNotification<any> = undefined;
+    let lastRenderedValue: any = undefined;
+
+    this.subscription.add(
+      this.letDirective.values$.subscribe((v) => {
```

Comment:
Let directive is optional. On line 120 you're checking if let directive is present, shouldn't you do it here also?